### ISSUE TITLE
Expose `keepdim` for `torchgeo.transforms.indices`

### ISSUE THREAD

**Issue by MathiasBaumgartinger** [microsoft/torchgeo#2329]
### Summary

Currently, none of the indices expose their inherited property `keepdim`. For some cases this might be useful.

### Rationale

Without the usage of `keepdim=True` on the augmentations, my CrossEntropyLoss fails with `RuntimeError: only batches of spatial targets supported (3D tensors) but got targets of dimension: [batch_size, 1, H, W]`, because an additional dimension is added.

### Implementation

Expose `keepdim: bool=False` in `__init()__`

**Comment by adamjstewart** [comment]
I've always wondered why `keepdim` isn't True by default, I can't think of any reason I would ever want it to be False.

Instead of explicitly adding a `keepdim` parameter, I would instead add `**kwargs` to all transforms and pass it directly to `super().__init__()`.

**Comment by MathiasBaumgartinger** [comment]
I agree, was quite flattered that some deep base augmentation class would for some reason add a completely unnecessary dimension. Anyway, some form of (by default `keepdim=True`) option and/or hint would be beneficial. Took me some time to debug that.

**Comment by adamjstewart** [comment]
Want to open a PR to add the `kwargs` forwarding? You could then show how to use `keepdim` in our [Transforms](https://torchgeo.readthedocs.io/en/stable/tutorials/transforms.html) tutorial.
